WordPress/gutenberg
# [RNMobile] Fix HTML to blocks conversion when no transformations are available

## What?


Fixes an issue related to converting HTML to blocks when using the unsupported HTML block in the native version.

This issue was introduced in https://github.com/WordPress/gutenberg/pull/55461 after enabling the _Convert to blocks_ option in the Classic block.

## Why?


Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6420.
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/6084.
Related to https://github.com/wordpress-mobile/gutenberg-mobile/issues/6416.

## How?


* Since the HTML block is not supported in the native version, a native-only condition has been added to `htmlToBlocks` so instead of using the HTML block, it will render the HTML block as an unsupported block.
* Incorporate into the native version the unit tests related to the raw block handling executed in the web version. This will ensure that processing raw blocks will work the same way between platforms. **However, as it can be checked in the test file, several of the test cases are currently failing in the native version. [This required to disable them temporarily until we address them in the future](https://github.com/WordPress/gutenberg/pull/56723#discussion_r1412463547).**
* [Add an extra test case to cover the case related to the issues solved by this PR](https://github.com/WordPress/gutenberg/pull/56723#discussion_r1412463856).

## Testing Instructions




1. Open the app and create a post.
2. Switch to HTML mode.
3. Paste the following HTML code:
```
<div>
<p>Hello world!</p>
</div>
```
4. Switch back to Visual mode.
5. Tap two times on the Classic block to present the bottom sheet.
6. Tap on _Convert to blocks_ option.
7. Observe that no exception/crash is produced.
8. Observe that the Classic block is converted to an HTML block (displayed as an unsupported block).
9. Edit the HTML block via UBE (Unsupported Block Editor).
10. Observe that the content of the HTML block matches the copied HTML from the third step.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A
